src/ser-ext-ondemand.ts

```diff
@@ -24,6 +24,7 @@ utils.checkDirectiveIsRegistrated($injector, qvangular, "", OnDemandDirectiveFac
 
 interface IVMScopeExtended extends utils.IVMScope<OnDemandExtension> {
     layout: ILayout;
+    vm: any;
```

Comment:
have to added this, later in your code you ask for scope.vm

```ts
    controller: ["$scope", function (scope: IVMScopeExtended) {

        //#region Logger
        logging.LogConfig.SetLogLevel("*", scope.layout.properties.loglevel);
        let logger = new logging.Logger("Main");
        //#endregion

        propertyScope = scope;
        scope.vm = new OnDemandExtension(scope);
    }]
```

and this crashes since vm is not defined, since the interface comes from you i simply added the property vm and set this to any.